protected_tags.go

```diff
@@ -0,0 +1,141 @@
+package gitlab
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// ProtectedTagsService handles communication with the protected tag methods of the GitLab API.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/protected_tags.html
+type ProtectedTagsService struct {
+	client *Client
+}
+
+// TagAccessDescription reperesents the access decription for a protected tag.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/protected_tags.html#list-protected-tags
+type TagAccessDescription struct {
+	AccessLevel            AccessLevelValue `json:"access_level"`
+	AccessLevelDescription string           `json:"access_level_description"`
+}
+
+// ProtectedTag represents a protected tag.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/protected_tags.html#list-protected-tags
+type ProtectedTag struct {
+	Name               string                  `json:"name"`
+	CreateAccessLevels []*TagAccessDescription `json:"create_access_levels"`
+}
+
+// ListProtectedTagsOptions represents available ListProtectedTags() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/protected_tags.html#list-protected-tags
+type ListProtectedTagsOptions ListOptions
+
+// ListProtectedTags returns a list of protected tags from a project.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/protected_tags.html#list-protected-tags
+func (s *ProtectedTagsService) ListProtectedTags(pid interface{}, opt *ListProtectedTagsOptions, options ...OptionFunc) ([]*ProtectedTag, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/protected_tags", url.QueryEscape(project))
+
+	req, err := s.client.NewRequest("GET", u, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var p []*ProtectedTag
```

Comment:
Maybe `pt` is a better resemblance for `ProtectedTag`? And then this should become `pts` to indicate is are multiple tags.